### ISSUE TITLE
TypeError: Cannot read property 'length' of undefined

### ISSUE THREAD

**Issue by wind4gis** [webpack/webpack-cli#1023]
**Describe the bug**

A 💯 way to do this is to provide your configuration via a GitHub gist and/or to use markdown when describing which commands you used when the error occurred! Be precise and clear in your description of the bug.

**What is the current behavior?**
when I run `webpack --mode development(or production)`, the webpack-cli shows `TypeError: Cannot read property 'length' of undefined` on the line `node_modules\webpack-cli\bin\cli.js:93`. my computer is win10

This is where you elaborate on the current behavior of the cli, how it behaves normally differing from the previous version/build you did. 

**To Reproduce**

1.just create a folder on win10
2.npm i -S webpack@next
3.npm i -S webpack-cli@latest
4.write "dev": "webpack --mode development" in package.json
5.npm run dev

**Expected behavior**

**Screenshots**
![image](https://user-images.githubusercontent.com/3963291/63011085-2ecea580-beba-11e9-881e-349e1d1e202f.png)

**Please paste the results of `webpack-cli info` here, and mention other relevant information**
![image](https://user-images.githubusercontent.com/3963291/63011134-44dc6600-beba-11e9-998e-f02944031b85.png)

**Additional context**
Add any other context about the problem here like linking to an similar issue you might think is the cause.

**Comment by evenstensberg** [comment]
Hi, thanks for submitting an issue. It's a bug, I've done a patch here:

https://github.com/webpack/webpack-cli/pull/1024

Which hopefully will land as a patch soon

**Comment by wind4gis** [comment]
@evenstensberg  It seems that there is still this bug

**Comment by evenstensberg** [comment]
Could you try out the #1024 branch and check if it fixes the bug?

**Comment by ryanspice** [comment]
I ran into this issue and pulled your branch it seems to be working now.

**Comment by ryanspice** [comment]
@evenstensberg not sure if this is to be expected but an empty webpack config object
![image](https://user-images.githubusercontent.com/3946233/63227173-dd703000-c1b1-11e9-8c3d-57cf794b9b0d.png) brings up the same error.

**Comment by ryanspice** [comment]
actually it might be something else, not quite sure, but trying to plug in my existing config I have seen this error pop up.

**Comment by ryanspice** [comment]
yeah im not sure why but when i tried to run it with a config now it was always giving me this error 

![image](https://user-images.githubusercontent.com/3946233/63227863-5757e780-c1b9-11e9-85ce-2e7830dd87a3.png)

it looks like _webpackConfigurationValidationErrors_ is returning undefined?? 

it runs if i modify a file

![image](https://user-images.githubusercontent.com/3946233/63227890-95eda200-c1b9-11e9-8b9f-8213cee07808.png)

**Comment by evenstensberg** [comment]
yes, options might be null / undefined, so the error is from validating the schema. Do you think you can figure out when the options obj is nil @ryanspice ?

**Comment by ryanspice** [comment]
@evenstensberg yeah it seems like _validateschemea.js_ on webpack@5.0.0-alpha.20 doesnt have a return

![image](https://user-images.githubusercontent.com/3946233/63228052-da7a3d00-c1bb-11e9-8b06-33dd47a40735.png)

**Comment by evenstensberg** [comment]
@sokra can we have that?

**Comment by ryanspice** [comment]
I created a bug report: https://github.com/webpack/webpack/issues/9591 

I was gonna create a branch and make a PR but then I realised I dont think i can do that there.

**Comment by sokra** [comment]
The `validateSchema` API changed a bit, since we are using `schema-util` now. It throws a `ValidationError` when validation failed. Also note that the `name` of the validation error changed from `WebpackOptionsValidationError` to `ValidationError`.

**Comment by evenstensberg** [comment]
Do you have a proposed way of using the schema with the new api? Not getting a message from validating the schema as of:

```js
const validateSchema = require("webpack").validateSchema;
const ValidationError = require("webpack").validate;

const webpackConfigurationValidationErrors = validateSchema(webpackConfigurationSchema, options);
if (webpackConfigurationValidationErrors.length) {
       const error = ValidationError(webpackConfigurationValidationErrors);
        console.error(error.message);
	process.exit(-1);
}
```

**Comment by sokra** [comment]
``` js
try {
  validateSchema(webpackConfigurationSchema, options);
} catch(error) {
  console.error(error.message);
  process.exit(-1);
}
```

**Comment by sokra** [comment]
If you want to support both version maybe like this: 

``` js
let error;
try {
  const errors = validateSchema(webpackConfigurationSchema, options);
  if(errors && errors.length > 0) {
    const WebpackOptionsValidationError = require("webpack").WebpackOptionsValidationError;
    error = new WebpackOptionsValidationError(errors);
  }
} catch(err) {
  error = err;
}

if(error) {
  console.error(error.message);
  process.exit(-1);
}
```

**Comment by ryanspice** [comment]
I'm not quite sure I understand why returning the value isn't appropriate.. It returns an empty array with no errors, would it not have an array populated with errors if there were some?
I probably just don't understand a fundamental part like how it would set the value if its not returning anything in the first place.

**Comment by willgriffiths** [comment]
Fixed by #1065. Landed in v3.3.8.
